graviti/schema/catalog_to_schema.py

```diff
@@ -0,0 +1,271 @@
+)
+from tensorbay.label.basic import SubcatalogBase, _LabelBase
+from tensorbay.label.supports import AttributesMixin, CategoriesMixin, MaskCategoriesMixin
+
+_SUFFIX_AND_FILE_TYPES = {
+    ".bmp": ("image", "file.Image"),
+    ".jpg": ("image", "file.Image"),
+    ".jpeg": ("image", "file.Image"),
+    ".png": ("image", "file.Image"),
+    ".gif": ("image", "file.Image"),
+    ".bin": ("point_cloud", "file.PointCloudBin"),
+    ".pcd": ("point_cloud", "file.PointCloud"),
+    ".ply": ("point_cloud", "file.PointCloud"),
+    ".las": ("point_cloud", "file.PointCloud"),
+    ".laz": ("point_cloud", "file.PointCloud"),
+    ".mp3": ("audio", "file.Audio"),
+    ".wav": ("audio", "file.Audio"),
+    ".wma": ("audio", "file.Audio"),
+    ".aac": ("audio", "file.Audio"),
+}
+
+_JSON_TYPE_TO_PORTEX_TYPE = {
+    "array": "array",
+    "boolean": "boolean",
+    "integer": "int32",
+    "number": "float32",
+    "string": "string",
+}
+
+_LABEL_PROCESSORS: Dict[str, Callable[..., Any]] = {}
+
+
+def _get_category_params(subcatalog: Union[CategoriesMixin, MaskCategoriesMixin]) -> Dict[str, Any]:
+    if not hasattr(subcatalog, "categories"):
+        return {}
+
+    category_params: Dict[str, Any] = {}
+    if isinstance(subcatalog, MaskCategoriesMixin):
+        category_params["categoryIds"] = subcatalog.get_index_to_category()
+    else:
+        category_params["categories"] = list(subcatalog.categories.keys())
+
+    category_delimiter = getattr(subcatalog, "category_delimiter", None)
+    if category_delimiter:
+        category_params["categoryDelimiter"] = category_delimiter
+
+    return category_params
+
+
+def _get_attribute_params(subcatalog: AttributesMixin) -> Dict[str, Any]:
+    if not hasattr(subcatalog, "attributes"):
+        return {}
+
+    attribute_params: List[Dict[str, Any]] = []
+    for attribute in subcatalog.attributes:
+        field: Dict[str, Any] = {"name": attribute.name}
+        if hasattr(attribute, "enum"):
+            field["type"] = "enum"
+            field["values"] = attribute.enum
+        else:
+            attribute_type = (
+                reduce(lambda a, b: a or b, attribute.type)
+                if isinstance(attribute.type, list)
+                else attribute.type
+            )
+            field["type"] = _JSON_TYPE_TO_PORTEX_TYPE[attribute_type]
+
+            if hasattr(attribute, "minimum"):
+                field["minimum"] = attribute.minimum
+            if hasattr(attribute, "maximum"):
+                field["maximum"] = attribute.maximum
+        attribute_params.append(field)
+    return {"attributes": attribute_params}
+
+
+def _get_category_and_attribute_params(subcatalog: Any) -> Dict[str, Any]:
+    return {**_get_category_params(subcatalog), **_get_attribute_params(subcatalog)}
+
+
+def _get_dtype(value: float, default_type: str = "int") -> Dict[str, str]:
+    dtype = type(value).__name__
+    return {"dtype": f"{dtype}32"} if dtype != default_type else {}
```

Comment:
this code is so hard.
can we use a dict to maintain the relationship of python type and portex type?